packcheck-remote.sh

```diff
@@ -0,0 +1,208 @@
+            return
+        fi
+    fi
+
+    mkdir -p "$dir" || exit 1
+
+    run_verbose "git clone $remote $dir" || exit 1
+
+    cd "$dir" || exit 1
+    run_verbose "git branch $rev" || exit 1
+
+    if test -n "$merge"
+    then
+        # This will fail if there are any merge conflicts
+        run_verbose "git merge -X theirs $merge --no-edit --commit" || exit 1
+    fi
+
+    show_step "Running packcheck"
+    # Run packcheck here
+    run_verbose "$PACKCHECK_EXE $packcheck_cli_opts"
+}
+
+# Arguments for the command line
+FORCE="false"
+REMOTE=""
+CHECKOUT=""
+MERGE=""
+DIRECTORY=""
+PACKCHECK_CLI_OPTS_ARR=()
+PACKCHECK_CLI_OPTS=""
+
+function run_help() {
+    local script
+    script=$(basename "$0")
+
+    echo
+    echo "USAGE: --option[=value]"
+    echo
+    echo "-f (or) --force: Puts the script in force mode. This is ideal for CIs."
+    echo "-h (or) --help:  Print help."
+    echo "--remote:        Repository to clone."
+    echo "--checkout:      Revision to checkout. Defaults to 'origin/master'."
+    echo "--merge:         Revision to merge in the checked out branch."
+    echo "--directory:     Directory to clone the repository into."
+    echo "                 Defaults to '$DEFAULT_DIRECTORY_NAME' under the present working directory."
+    echo
+    echo "All the arguments after '--' are passed to packcheck"
+    echo
+
+    echo
+    echo "EXAMPLE:"
+    echo
+    echo "$script --force \\"
+    echo "        --remote=https://github.com/user/repo \\"
+    echo "        --checkout=origin/master \\"
+    echo "        --merge=origin/branch \\"
+    echo "        --directory=./repo.packcheck \\"
+    echo "        -- cabal-v2 GHCVER=8.8.3"
+    echo
+}
+
+# Program entry point
+for i in "$@"
+do
+    case $i in
+        -f|--force)
+            FORCE="true"
+            shift
+            ;;
+        -h|--help)
+            run_help
+            exit 0
+            ;;
+        --remote=*)
+            REMOTE="${i#*=}"
+            shift
+            ;;
+        --checkout=*)
+            CHECKOUT="${i#*=}"
+            shift
+            ;;
+        --merge=*)
+            MERGE="${i#*=}"
+            shift
+            ;;
+        --directory=*)
+            DIRECTORY="${i#*=}"
+            shift
+            ;;
+        --)
+            shift
+            PACKCHECK_CLI_OPTS_ARR=("$@")
+            break
+            ;;
+        *)
+            echo "Unknown argument to packcheck-remote"
+            run_help
+            exit 1
+            ;;
+    esac
+done
+
+for i in "${PACKCHECK_CLI_OPTS_ARR[@]}"
+do
+    case $i in
+        *=*)
+            key=${1%%=*}
+            val=${1#*=}
+            PACKCHECK_CLI_OPTS="$PACKCHECK_CLI_OPTS $key=\"$val\""
+            shift
+            ;;
+        *)
+            PACKCHECK_CLI_OPTS="$PACKCHECK_CLI_OPTS $i"
+            shift
+            ;;
+    esac
+done
+
+try_git_clone_and_merge "$REMOTE" "$CHECKOUT" "$MERGE" "$DIRECTORY" "$FORCE" "$PACKCHECK_CLI_OPTS"
```

Comment:
Can't we just pass `"$*"` instead of processing `$@`?

Reply:
This does not work. `a="b c d"` would become `a=b c d`.